src/onegov/form/display.py

```diff
@@ -121,6 +122,78 @@ def __call__(self, field: 'Field') -> Markup:
         ).format(url=field.data)
 
 
+@registry.register_for('VideoURLField')
+class VideoURLFieldRenderer(BaseRenderer):
+    """
+    Renders a video url. Embeds the video if in case of vimeo or
+    youtube otherwise just displays the url as a link.
+    """
+
+    video_template = """
+        <div class="video">
+            <h2 i18n:translate="" class="visually-hidden">Video</h2>
+            <div class="videowrapper">
+                <iframe allow="fullscreen"
+                frameborder="0" src="${url}"></iframe>
+            </div>
+        </div>
+    """
+    url_video_template = """<a href=${url}>${url}</a>"""
+
+    video_page = PageTemplate(video_template)
+    url_video_page = PageTemplate(url_video_template + video_template)
+
+    def __call__(self, field: 'Field', **kwargs: Any) -> Markup:
+        url = None
+        data = self.escape(field.data)
+
+        if kwargs:
```

Comment:
@Daverball I did not find a good way to distinguish between editing and displaying the video url.
My goal is to show the 'plain url' in a `StringField` when editing the directory entry and the embedded video otherwise.

Reply:
TODO: Support multiple videos

Reply:
Field display is strictly for display, it should never be editable, so it does not make sense to render an input field here.

Reply:
You'll want to create a separate Widget for rendering the input, but considering the widget is just a regular url field anyways, you don't need a custom widget.